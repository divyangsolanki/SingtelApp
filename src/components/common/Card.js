import React from 'react';
import {Text, View, StyleSheet, TouchableOpacity} from 'react-native';

import FlipCard from 'react-native-flip-card';

const Card = props => {
  return (
    <View style={styles.cardContainer} testID={props.testID}>
      <FlipCard style={styles.card} flip={props.isFlipped} clickable={true}>
        <TouchableOpacity style={styles.face} onPress={props.onPress}>
          <Text>Card </Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.back} onPress={props.onPress}>
          <Text>{props.val}</Text>
        </TouchableOpacity>
      </FlipCard>
    </View>
  );
};

const styles = StyleSheet.create({
  cardContainer: {
    margin: 10,
    borderColor: 'blue',
    borderWidth: 2,
    flexWrap: 'wrap',
  },
  card: {
    width: 100,
    height: 100,
  },
  face: {
    flex: 1,
    backgroundColor: '#2ecc71',
    justifyContent: 'center',
    alignItems: 'center',
  },
  back: {
    flex: 1,
    backgroundColor: '#f1c40f',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Card;
