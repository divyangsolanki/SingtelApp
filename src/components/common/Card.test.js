import React from 'react';
import {render} from '@testing-library/react-native';
import Card from './Card';

describe('Test Card', () => {
  it('Card', () => {
    const wrapper = render(
      <Card onPress={() => {}} isFlipped={false} val={4} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
