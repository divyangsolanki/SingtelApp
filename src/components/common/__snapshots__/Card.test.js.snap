// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Card 1`] = `
<View
  style={
    Object {
      "borderColor": "blue",
      "borderWidth": 2,
      "flexWrap": "wrap",
      "margin": 10,
    }
  }
/>
`;
